import "type.proto";
package pbnavitia;

message PlacesRequest {
    required string q           = 1;
    repeated NavitiaType types  = 2;
    required int32 depth        = 3;
    required int32 count        = 4;
    repeated string admin_uris  = 5;
}

message NextStopTimeRequest {
    required string departure_filter = 1;
    required string arrival_filter   = 2;
    required string from_datetime    = 3;
    required int32 duration          = 4;
    required int32 depth             = 5;
    required int32 nb_stoptimes      = 7;
    optional int32 interface_version = 8;
    required int32 start_page        = 9;
    required int32 count             = 10;
}

message StreetNetworkParams{
    optional string origin_mode         = 1;
    optional string destination_mode    = 2;
    optional double walking_speed       = 3;
    optional int32 walking_distance     = 4;
    optional double bike_speed          = 5;
    optional int32 bike_distance        = 6;
    optional double car_speed           = 7;
    optional int32 car_distance         = 8;
    optional double vls_speed           = 9;
    optional int32 vls_distance         = 10;
    optional string origin_filter       = 11;
    optional string destination_filter  = 12;
}


message JourneysRequest {
    required string origin                              = 1;
    optional string destination                         = 2;
    repeated string datetimes                           = 3;
    required bool clockwise                             = 4;
    repeated string forbidden_uris                      = 5;
    required int32 max_duration                         = 6;
    required int32 max_transfers                        = 7;
    optional StreetNetworkParams streetnetwork_params   = 8;
}

message PlacesNearbyRequest {
    required string uri         = 1;
    required double distance    = 2;
    repeated NavitiaType types  = 3;
    required int32 depth        = 4;
    required int32 count        = 5;
    required int32 start_page   = 6;
}

message PTRefRequest {
    required NavitiaType requested_type = 1;
    required string filter              = 2;
    required int32 depth                = 3;
    required int32 start_page           = 4;
    required int32 count                = 5;
}

message Request {
    required API requested_api                      = 1;

    optional PlacesRequest places                   = 2;
    optional NextStopTimeRequest next_stop_times    = 3;
    optional PlacesNearbyRequest places_nearby      = 4;
    optional JourneysRequest journeys               = 5;
    optional PTRefRequest ptref                     = 6;

}
